### PR TITLE
Quickstart solves #190 #163

### PR DESCRIPTION
**Pull request by kordc**
[add quickstart modules](https://github.com/SebChw/Actually-Robust-Training/pull/186/commits/9a4a05654e10ade7bdc2580bd4fccb4fbed8c283) - As suggested. I placed few Dummy modules inside art now quickstart work withouth any additional effort.
[organize tests](https://github.com/SebChw/Actually-Robust-Training/pull/186/commits/3d976356a279425892391b600e52e41dc2cc9ef1) - Since I wanted to write another test for quickstart I made thing reusable

[allow custom metric name inside check](https://github.com/SebChw/Actually-Robust-Training/pull/186/commits/e6c2db0f7eca017ab7016d04d1058b3042cfffd5), [make checks optional](https://github.com/SebChw/Actually-Robust-Training/pull/186/commits/5540243fcf5f70f5eb330f51854afe3da876737c), 
make log params optional - These doesn't break logic of ART and everything written before works. I just thougt that someone Could potentially cherrypick only some features of ART

[make metric calculator optional](https://github.com/SebChw/Actually-Robust-Training/pull/186/commits/9d685aa734c0bb2f162fbd79376464c374db14f5) - This is for someone that just want to wrap their lightning modules with steps and checks. Unfortunatelly this required a little bit effort. I put entire metric calculator handling on the step side as now it can be None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,50 +8,203 @@ https://readthedocs.org/projects/actually-robust-training/badge/?version=latest&
...
 ## Installation
 To get started, install ART package using:
 ```sh
 pip install art-training
 ```
+
+## Quickstart
+
+### Steps
+The basic idea behind ART is to split your deep learning pipeline into a series of _steps_:
```
Comment (kordc) [inline README.md]: I'd write 1-2 more lines about it

File: README.md [hunk 2]
```diff
@@ -8,50 +8,203 @@ https://readthedocs.org/projects/actually-robust-training/badge/?version=latest&
...
+Step: Overfit One Batch, Model: YourLightningModule, Passed: True. Results:
+         loss-train: 0.00035009183920919895
+         accuracy-train: 0.800000011920929
+Step: Overfit, Model: YourLightningModule, Passed: True. Results:
+         loss-train: 2.1859991550445557
+         accuracy-train: 0.75
+         loss-valid: 3.5187878608703613
+```
+
+[Explore all available Steps](not-existing-link)
```
Comment (kordc) [inline README.md]: Will be done by #129

File: README.md [hunk 3]
```diff
@@ -8,50 +8,203 @@ https://readthedocs.org/projects/actually-robust-training/badge/?version=latest&
...
+
+```
+Check failed for step: Check Loss On Init. Reason: Score 16.067102432250977 is not equal to 2.3
+Summary:
+Step: Check Loss On Init, Model: YourLightningModule, Passed: False. Results:
+         loss-valid: 16.067102432250977
+         accuracy-train: 0.75
+```
+
+### Debug your Neural Network
```
Comment (kordc) [inline README.md]: I'd remove it from the quickstart, it's not important for the starting the experiment. I'd leave it for tutorials - we could make another one, as we were discussing in #72

File: README.md [hunk 4]
```diff
@@ -8,50 +8,203 @@ https://readthedocs.org/projects/actually-robust-training/badge/?version=latest&
...
+    art_decorate([(model_class, "forward")], BatchSaver())
+
+    project = ArtProject(name="quickstart", datamodule=datamodule)
+    # For this to work we should allow providing empty checks. And just print a warning... man use checks.
+    project.add_step(CheckLossOnInit(model_class))
+    project.run_all()
+
+```
+
+### Get control over what is being calculated
```
Comment (kordc) [inline README.md]: Also it may be too huge for quickstart? Just loudly thinking, we introduce it in tutorials.

**Review by trebacz626 — DISMISSED** [review]

**Review by trebacz626 — APPROVED** [review]

**Pull request merged**